tests/deserialize/test_validators.py

```diff
@@ -105,3 +105,23 @@ class Schema(halogen.Schema):
     with pytest.raises(halogen.exceptions.ValidationError) as err:
         Schema.deserialize({"attr": 3})
     assert err.value.errors[0].errors == ["3 is greater than maximum value 2"]
+
+
+def test_one_of():
+    class Schema(halogen.Schema):
+
+        attr = halogen.Attr(
+            halogen.types.Type(
+                validators=[
+                    halogen.validators.OneOf(
+                        [1, 2, 3]
+                    )
+                ]
+            )
+        )
+
+    with pytest.raises(halogen.exceptions.ValidationError) as err:
+        Schema.deserialize({"attr": 4})
+    assert err.value.errors[0].errors == ['"4" is not a valid choice']
+
+    Schema.deserialize({"attr": 3})
```

Comment:
assert the result here

Reply:
The result is not being tested here, the validation is. Adding an assert on the de-serialize might be confusing and obfuscating.